internal/logger/deployment.go

```diff
@@ -3,17 +3,41 @@ package logger
 import (
 	"context"
 	"fmt"
-
+	"github.com/observatorium/loki-benchmarks/internal/config"
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
 	"sigs.k8s.io/controller-runtime/pkg/client"
-
-	"github.com/observatorium/loki-benchmarks/internal/config"
 )
 
 func Deploy(c client.Client, cfg *config.Logger, scenarioCfg *config.Writers, pushURL string) error {
+	var (
+		args               []string
+		addDefaultPushURL  = true
+		addDefaultTenantID = true
+	)
+
+	for k, v := range scenarioCfg.Args {
+		switch v {
+		case "#pushURL":
+			args = append(args, fmt.Sprintf("--%s=%s", k, pushURL))
+			addDefaultPushURL = false
+		case "#tenantID":
+			args = append(args, fmt.Sprintf("--%s=%s", k, cfg.TenantID))
+			addDefaultTenantID = false
```

Comment:
You don't need `#pushURL` and `#tenantID` actually, you can use the keys `url` and `tenantID`. If the keys exist, you can override the defaults.

Reply:
@periklis The point is to allow using overridden keys ... so for example use the key `blablaurl` instead of the key `url` so if someone wants to use the key  `blablaurl` he will configure  `blablaurl:  #pushURL` and the way for us to know that he configured that and that we do not need to add the key `url` is by observing the value, right?

Reply:
Using `#pushURL` is not an obvious usage. Let's keep a README.md not as: `Overriding url and tenant requires that the logger implementation provides such named CLI flags.`

Reply:
Fixed
DONE